packages/vaadin-grid/test/array-data-provider.test.js

```diff
@@ -250,3 +251,40 @@ describe('invalid paths', () => {
     });
   });
 });
+
+describe('items with a custom data provider', () => {
+  let grid;
+  const dataProvider = (params, callback) => callback([{ value: 'foo' }], 1);
+  const items = [{ value: 'bar' }];
+
+  beforeEach(() => {
+    grid = fixtureSync(`
+      <vaadin-grid>
+        <vaadin-grid-column path="value"></vaadin-grid-column>
+      </vaadin-grid>
+    `);
+  });
+
+  it('use the items array', () => {
+    grid.dataProvider = dataProvider;
+    grid.items = items;
+    flushGrid(grid);
+    expect(getBodyCellContent(grid, 0, 0).textContent).to.equal('bar');
+    expect(grid.dataProvider).not.to.equal(dataProvider);
+  });
+
+  it('should use the data provider', () => {
```

Comment:
**nit:** Since there are 2 types of data providers, I would suggest clarifying which data provider is meant here.

```suggestion
  it('should use the custom data provider', () => {
```